### PR TITLE
Provide Write(), Stderr(), Printf() methods on App

### PR DESCRIPTION
**Pull request by Matir**
This allows consumers of the library to write in a manner that refreshes
the prompt because it goes through the Readline library.  This is useful
for background tasks or notifications to correctly redraw your screen.

### CODE REVIEW

**Comment by Matir** [comment]
This also implements `Printf()` and `Println()` in the style of #7, as well as the request to use the implemented functions for all the uses of printing within grumble.

I also updated the examples to use the same approach and ran them to verify that everything works as expected.

**Comment by skaldesh** [comment]
Thank you for your contribution! I checked it out swiftly and it looks good so far. If @r0l1 gives his OK, we can merge it!

File: app.go [hunk 1]
```diff
@@ -167,6 +179,23 @@ func (a *App) SetPrintASCIILogo(f func(a *App)) {
...
+func (a *App) Write(p []byte) (int, error) {
+	if a.rl != nil {
+		return a.rl.Stdout().Write(p)
+	}
+	return os.Stdout.Write(p)
+}
+
+// Stderr returns a writer to Stderr, using readline if available.
+// Note that calling before Run() will return a different instance.
+func (a *App) Stderr() io.Writer {
```
Comment (r0l1) [inline app.go]: Please also specify a Stdout method on the APP, if Stderr is provided.

File: app.go [hunk 2]
```diff
@@ -167,6 +179,23 @@ func (a *App) SetPrintASCIILogo(f func(a *App)) {
 	}
 }
 
+// Write to the underlying output, using readline if available.
+func (a *App) Write(p []byte) (int, error) {
+	if a.rl != nil {
```
Comment (r0l1) [inline app.go]: Make use of a.Stdout()

File: functions.go [hunk 3]
```diff
@@ -189,24 +189,28 @@ func defaultPrintCommandHelp(a *App, cmd *Command, shell bool) {
...
+		a.Printf("%s\n", columnize.Format(output, config))
 	}
 
-	fmt.Println()
+	a.Println()
 }
 
-func headlinePrinter(c *Config) func(v ...interface{}) (int, error) {
-	if c.NoColor || c.HelpHeadlineColor == nil {
-		return fmt.Println
```
Comment (r0l1) [inline functions.go]: Please use a.Println

File: app.go [hunk 4]
```diff
@@ -167,6 +179,23 @@ func (a *App) SetPrintASCIILogo(f func(a *App)) {
...
+		return a.rl.Stdout().Write(p)
```
Comment (r0l1) [inline app.go]: Make use of a.Stdout()

File: app.go (hunk 1 above)
Comment (r0l1) [inline app.go]: Please also implement a Stdout method on the APP, if Stderr is provided.

**Comment by Matir** [comment]
I believe I've addressed the comments, please take another look!

File: functions.go [hunk 5]
```diff
@@ -200,7 +200,7 @@ func defaultPrintCommandHelp(a *App, cmd *Command, shell bool) {
 func headlinePrinter(a *App) func(v ...interface{}) (int, error) {
 	if a.config.NoColor || a.config.HelpHeadlineColor == nil {
 		return func(v ...interface{}) (int, error) {
```
Comment (skaldesh) [inline functions.go]: Instead of wrapping a.Println in a func(v ...interface{}) (int, error) here, you can directly return a.Println!

Comment (Matir) [reply functions.go]: Yup, that was a stupid one.  Fixing it now.  :)

**Pull request merged**